tests/posix/pthread_rwlock/src/posix_rwlock.c

```diff
@@ -80,7 +80,8 @@ static void test_rw_lock(void)
 
 	/* Creating N premptive threads in increasing order of priority */
 	for (i = 0; i < N_THR; i++) {
-		pthread_attr_init(&attr[i]);
+		zassert_false(pthread_attr_init(&attr[i]),
```

Comment:
actually, turn this into zassert_equal and expect 0, true/false might be misleading here, the call should return 0 on success and nonzero error numbers.